e2e/test/scenarios/question/reproductions/19893-joining-same-table-and-field-multiple-times.cy.spec.js

```diff
@@ -0,0 +1,118 @@
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { restore } from "e2e/support/helpers";
+
+const { PRODUCTS, PRODUCTS_ID } = SAMPLE_DATABASE;
+
+const QUESTION_1 = {
+  name: "Q1",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [["count"]],
+    breakout: [["field", PRODUCTS.CATEGORY, { "base-type": "type/Text" }]],
+  },
+};
+
+const QUESTION_2 = {
+  name: "Q2",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [
+      ["sum", ["field", PRODUCTS.PRICE, { "base-type": "type/Float" }]],
+    ],
+    breakout: [["field", PRODUCTS.CATEGORY, { "base-type": "type/Text" }]],
+  },
+};
+
+describe("issue 19893", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should display correct join source table when joining visited questions (metabase#19893)", () => {
+    cy.createQuestion(QUESTION_1, {
+      wrapId: true,
+      idAlias: "questionId1",
+      visitQuestion: true,
+    });
+    cy.createQuestion(QUESTION_2, {
+      wrapId: true,
+      idAlias: "questionId2",
+      visitQuestion: true,
+    });
+
+    cy.then(function () {
+      const { questionId1, questionId2 } = this;
+
+      createQ1PlusQ2Question(questionId1, questionId2).then(
+        ({ body: question }) => {
+          cy.visit(`/question/${question.id}/notebook`);
+        },
+      );
+    });
+
+    assertQ1PlusQ2Joins();
+  });
+
+  it.skip("should display correct join source table when joining non-visited questions (metabase#19893)", () => {
+    cy.createQuestion(QUESTION_1, { wrapId: true, idAlias: "questionId1" });
+    cy.createQuestion(QUESTION_2, { wrapId: true, idAlias: "questionId2" });
+
+    cy.then(function () {
+      const { questionId1, questionId2 } = this;
```

Comment:
I haven't seen this pattern before. Not sure if it makes sense to me. Wdyt about
```
cy.get("@questionId1").then(questionId1 => {
  cy.get("@questionId2").then(questionId2 => {
    createQ1PlusQ2Question(questionId1, questionId2)....
  });
});
```

Reply:
I saw @WiNloSt use this pattern a few times, and I like it because it helps to save `n - 1` levels of indentation, where `n` is the count of variables you want to use. It makes code inside the callback shorter and more readable.

Reply:
Yeah, I've used this in a few places.

In an instance there are 3 variables and nesting `cy.get` would be very hard to read.

https://github.com/metabase/metabase/blob/7a202816024a0174c4a083ffee373087370cd43b/e2e/test/scenarios/dashboard/reproductions/13736-dashboard-should-work-if-some-cards-are-broken.cy.spec.js#L41-L44